### ISSUE TITLE
[Suggestion] The blood magic integration seems underpowered.

### ISSUE THREAD

**Issue by TheInfamousAlk** [Ipsis/Woot#418]
So how I see it, woot is an endgame mod. It prevents me from having to make massive mob farms on my server, and provides a great way of automating mob drops that gives better drops per second than many other mob farming, is highly automatable and customizable, and most importantly, eliminates entity lag from servers. (which I am playing on/own)

However, I am having the opposite experience with blood collection. I have 1 factory with max mass, rate, altar and crystal upgrades on it. I have a max tier altar with 2/3rds of the upgrades as sacrifice and the rest as speed. (This is the same setup I had in a previous world without woot.)

The blood generation is nowhere near where I need it to be. Previously, I would put a well of suffering with a room full of witches below my altar (like maybe 30 or more witches or something) and the witches healed themselves so it was infinite blood. And that took up less space than the factory does, and required way fewer resources. I only had to catch a single witch in a mob imprisonment tool and clone it a bunch into the witch room then close it off and that was that, permanent blood generation.

With the expensive factory and upgrades, plus the ritual, it generates far less than half of what the manual ritual generates. I could probably fit 1 more factory below the altar, effectively doubling the blood output, but that still wouldn't generate as much as a witch prison. It might be possible to make 2 more above the altar, but at that point why even make factories? I could just use 30-40 witches. But with that comes lag, and I don't want to do that to my server.

The whole point of high blood output is so I can put some more speed runes in there and do some high tier crafting and not have to worry about not having enough blood, or so I can trigger tons of rituals at once or some sigil stuff, and not worry that my blood orb won't be filling up quickly enough. The idea of "not worrying about running out of x" is, I believe, the best part about endgame setups from tons of mods.

My point is, I think the blood output should be drastically increased so the factory is at least on par with, preferably much better than a manual witch blood farm, so it matches its status as a powerful but expensive endgame mob farm mod that it achieves in all other areas of the mod.

**Comment by Ipsis** [comment]
All the Blood Magic integration is configurable, so any modpack maker or user can increase or decrease the effectiveness of the various upgrades.

I ship the default config such that it is an alternative to the main Blood Magic rituals, rather than trying to one-up the BloodMagic version, however people are free to change that via the config if they see fit.

Is there something missing from the configuration options that doesn't allow the flexibility that you need?

**Comment by TheInfamousAlk** [comment]
No, I can configure it, but I'm sure 99% of people don't want to or care to configure it and will use the default configuration, and probably just decide to not use woot in favor of a witch farm. (_Also I'm not sure if every end user on a server would have to change their config to match the server's. If everyone does, unless a pack ships with the edit it will be difficult to get all of your users to edit the config, and annoying for end users who aren't very tech savvy_) 

I just feel like upping the effectiveness default would make a positive impact on the experience of many people who attempt to use woot and expect it to be at least as good as a "vanilla blood magic" solution. Because I know that I assumed from the start that since woot is "better" than most other mob farming methods, and more expensive than most, it would also be a better blood farm.

Because as it is now, in my opinion, the default setting for woot factory blood generation is a more expensive downgrade from a regular well of suffering ritual, since you can put TONS of mobs in range of the ritual that self heal and never die or despawn, and the amount of mobs (from mass and rate upgrades) in a woot factory is limited.

**Comment by TheInfamousAlk** [comment]
One of the most popular blood farms is villagers with an external healing source or witches that heal themselves, because I believe that "humans" like villagers and witches give more blood per sacrifice. Maybe a possible addition to woot would be some sort of "prison" upgrade or "life support" upgrade something that, when you put down a witch or villager core, would emulate the blood harvesting from a large number of them, but not drop anything because you're not really killing them. And with each tier it could add more theoretical witches or villagers to the "group" to be farmed at once.

That way, if you put down a normal factory with the blood upgrade, and something like zombies, it would kill them and harvest the amount of blood that it does now, which is a good representation of a vanilla blood farm that spawns and kills mobs. But if you used the villager or witch core with the "prison, etc" upgrade, it would be a good representation of the amount of blood you get from a self healing well of suffering that has non-undead humanoids in it.

**Comment by Ipsis** [comment]
Changing the defaults I wont be doing unfortunately, but I do like your second idea of an upgrade.

I'll have to think about how it would work because as with most things, it sounds simple but there is always a catch when it comes to implementation.

**Comment by Ipsis** [comment]
Sorry, this isn't going to make it into 1.12.
Just haven't had the time to do anything with it.
I'll move it into 1.14 and add it when I add the eventual/reworked BloodMagic support to Woot.

**Comment by TheInfamousAlk** [comment]
That's cool, take your time! All the mods 4 is already in 1.14 and going strong, so I hope that's where I'll see this mod first!
